cli/cmd/tap.go

```diff
@@ -140,38 +191,43 @@ func newCmdTap() *cobra.Command {
 	cmd.PersistentFlags().StringVar(&options.path, "path", options.path,
 		"Display requests with paths that start with this prefix")
 	cmd.PersistentFlags().StringVarP(&options.output, "output", "o", options.output,
-		"Output format. One of: wide")
+		fmt.Sprintf("Output format. One of: \"%s\", \"%s\"", wideOutput, jsonOutput))
 
 	return cmd
 }
 
-func requestTapByResourceFromAPI(w io.Writer, k8sAPI *k8s.KubernetesAPI, req *pb.TapByResourceRequest, wide bool) error {
-	var resource string
-	if wide {
-		resource = req.GetTarget().GetResource().GetType()
-	}
-
+func requestTapByResourceFromAPI(w io.Writer, k8sAPI *k8s.KubernetesAPI, req *pb.TapByResourceRequest, options *tapOptions) error {
 	reader, body, err := tap.Reader(k8sAPI, req, 0)
 	if err != nil {
 		return err
 	}
 	defer body.Close()
 
-	return renderTap(w, reader, resource)
+	return writeTapEventsToBuffer(w, reader, req, options)
 }
 
-func renderTap(w io.Writer, tapByteStream *bufio.Reader, resource string) error {
-	tableWriter := tabwriter.NewWriter(w, 0, 0, 0, ' ', tabwriter.AlignRight)
-	err := writeTapEventsToBuffer(tapByteStream, tableWriter, resource)
+func writeTapEventsToBuffer(w io.Writer, tapByteStream *bufio.Reader, req *pb.TapByResourceRequest, options *tapOptions) error {
+	writer := tabwriter.NewWriter(w, 0, 0, 0, ' ', tabwriter.AlignRight)
```

Comment:
`tabwriter` doesn't make much sense for json output. also the default and wide outputs are not outputting delimited output either, so we should be safe to remove it:
```diff
 func writeTapEventsToBuffer(w io.Writer, tapByteStream *bufio.Reader, req *pb.TapByResourceRequest, options *tapOptions) error {
-       writer := tabwriter.NewWriter(w, 0, 0, 0, ' ', tabwriter.AlignRight)
-
        var err error
        switch options.output {
        case "":
-               err = renderTapEvents(tapByteStream, writer, renderTapEvent, "")
+               err = renderTapEvents(tapByteStream, w, renderTapEvent, "")
        case wideOutput:
                resource := req.GetTarget().GetResource().GetType()
-               err = renderTapEvents(tapByteStream, writer, renderTapEvent, resource)
+               err = renderTapEvents(tapByteStream, w, renderTapEvent, resource)
        case jsonOutput:
-               err = renderTapEvents(tapByteStream, writer, renderTapEventJSON, "")
+               err = renderTapEvents(tapByteStream, w, renderTapEventJSON, "")
        }
        if err != nil {
                return err
        }
-       writer.Flush()

        return nil
 }

-func renderTapEvents(tapByteStream *bufio.Reader, w *tabwriter.Writer, render renderTapEventFunc, resource string) error {
+func renderTapEvents(tapByteStream *bufio.Reader, w io.Writer, render renderTapEventFunc, resource string) error {
```